qiskit/circuit/quantumcircuit.py

```diff
@@ -1445,60 +1417,32 @@ def qasm(self, formatted=False, filename=None, encoding=None):
             if instruction.name == "measure":
                 qubit = qargs[0]
                 clbit = cargs[0]
-                string_temp += "{} {} -> {};\n".format(
+                string_temp += "{} {} -> {};\n".forma(
                     instruction.qasm(),
                     bit_labels[qubit],
                     bit_labels[clbit],
                 )
-
-            # If instruction is a root gate or a root instruction (in that case, compositive)
-
-            elif (
-                type(instruction)
-                in [
-                    Gate,
-                    Instruction,
-                ]
-                or (isinstance(instruction, ControlledGate) and instruction._open_ctrl)
-            ):
-                if instruction not in existing_composite_circuits:
-                    if instruction.name in existing_gate_names:
-                        old_name = instruction.name
-                        instruction.name += "_" + str(id(instruction))
-
-                        warnings.warn(
-                            "A gate named {} already exists. "
-                            "We have renamed "
-                            "your gate to {}".format(old_name, instruction.name)
-                        )
-
-                    # Get qasm of composite circuit
-                    qasm_string = self._get_composite_circuit_qasm_from_instruction(instruction)
-
-                    # Insert composite circuit qasm definition right after header and extension lib
-                    string_temp = string_temp.replace(
-                        self.extension_lib, f"{self.extension_lib}\n{qasm_string}"
-                    )
-
-                    existing_composite_circuits.append(instruction)
-                    existing_gate_names.append(instruction.name)
-
-                # Insert qasm representation of the original instruction
-                string_temp += "{} {};\n".format(
-                    instruction.qasm(),
-                    ",".join([bit_labels[j] for j in qargs + cargs]),
-                )
             else:
-                string_temp += "{} {};\n".format(
+                # decompose gate using definitions if they are not defined in OpenQASM2
+                if instruction.name not in existing_gate_names:
+                    if instruction not in existing_composite_circuits:
+                        custom_instr_names = [
+                            instruction.name for instruction in existing_composite_circuits
+                        ]
+                        if instruction.name in custom_instr_names:
```

Comment:
Nested conditionals can be a combined with `and`:
```suggestion
                if instruction.name not in existing_gate_names and instruction not in existing_composite_circuits and instruction.name in [instruction.name for instruction in existing_composite_circuits]:
```